Source/automap.cpp

```diff
@@ -616,49 +779,79 @@ void SearchAutomapItem(const Surface &out, const Displacement &myPlayerOffset, i
 			if (!highlightTile({ i, j }))
 				continue;
 
-			int px = i - 2 * AutomapOffset.deltaX - ViewPosition.x;
-			int py = j - 2 * AutomapOffset.deltaY - ViewPosition.y;
+			DrawDiamond(out, GetAutomapScreen(i, j, myPlayerOffset), MapColorsItem);
+		}
+	}
+}
 
-			Point screen = {
-				(myPlayerOffset.deltaX * AutoMapScale / 100 / 2) + (px - py) * AmLine(16) + gnScreenWidth / 2,
-				(myPlayerOffset.deltaY * AutoMapScale / 100 / 2) + (px + py) * AmLine(8) + (gnScreenHeight - GetMainPanel().size.height) / 2
-			};
+void DrawAutomapObject(const Surface &out, const Displacement &myPlayerOffset, tl::function_ref<bool(Point position)> highlightTile)
+{
+	for (int i = 0; i < MAXDUNX; i++) {
+		for (int j = 0; j < MAXDUNY; j++) {
+			MapExplorationType explorationType = static_cast<MapExplorationType>(AutomapView[clamp(((i - 16) / 2), 0, DMAXX - 1)][clamp(((j - 16) / 2), 0, DMAXY - 1)]);
 
-			if (CanPanelsCoverView()) {
-				if (IsRightPanelOpen())
-					screen.x -= 160;
-				if (IsLeftPanelOpen())
-					screen.x += 160;
-			}
-			screen.y -= AmLine(8);
-			DrawDiamond(out, screen, MapColorsItem);
+			if (!highlightTile({ i, j }) || explorationType == MAP_EXP_NONE || !IsAnyOf(ObjectAtPosition({ i, j })._otype, OBJ_BLINDBOOK, OBJ_BLOODBOOK, OBJ_BOOK2R, OBJ_CRUX1, OBJ_CRUX2, OBJ_CRUX3, OBJ_L5BOOKS, OBJ_L5LEVER, OBJ_LAZSTAND, OBJ_LEVER, OBJ_MCIRCLE1, OBJ_MCIRCLE2, OBJ_MUSHPATCH, OBJ_PEDESTAL, OBJ_SIGNCHEST, OBJ_SLAINHERO, OBJ_STAND, OBJ_STORYBOOK, OBJ_SWITCHSKL, OBJ_WARARMOR, OBJ_WARWEAP))
+				continue;
+
+			DrawDiamond(out, GetAutomapScreen(i, j, myPlayerOffset), MapColorsObject);
 		}
 	}
 }
 
-/**
- * @brief Renders an arrow on the automap, centered on and facing the direction of the player.
- */
-void DrawAutomapPlr(const Surface &out, const Displacement &myPlayerOffset, int playerId)
+void DrawAutomapMissile(const Surface &out, const Displacement &myPlayerOffset)
 {
-	int playerColor = MapColorsPlayer + (8 * playerId) % 128;
+	for (int i = 0; i < MAXDUNX; i++) {
+		for (int j = 0; j < MAXDUNY; j++) {
+			MapExplorationType explorationType = static_cast<MapExplorationType>(AutomapView[clamp(((i - 16) / 2), 0, DMAXX - 1)][clamp(((j - 16) / 2), 0, DMAXY - 1)]);

```

Comment:
We no longer have `clamp`, this should be `std::clamp` in `development`.
Please `git rebase development` and retarget the PR.